azurerm/internal/services/datamigration/resource_arm_data_migration_service.go

```diff
@@ -0,0 +1,247 @@
+package datamigration
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/Azure/azure-sdk-for-go/services/datamigration/mgmt/2018-04-19/datamigration"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDataMigrationService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDataMigrationServiceCreate,
+		Read:   resourceArmDataMigrationServiceRead,
+		Update: resourceArmDataMigrationServiceUpdate,
+		Delete: resourceArmDataMigrationServiceDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"virtual_subnet_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"sku_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"kind": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      "Cloud",
+				ValidateFunc: validation.StringInSlice([]string{"Cloud"}, false),
+			},
+
+			"delete_running_tasks": {
```

Comment:
THis isn't; force new and is not in the update function? should this be in tf?

Reply:
After a reconsideration, I think we can just hide this property from user and force it as `true`. The meaning of this property is:

> deleteRunningTasks - delete the resource even if it contains running tasks

This is always guaranteed in terraform if **task** is another resource which depends on **data migration service**. So terraform has already handled it properly. For those **task** created out-of-bound, we can just force delete them.